### PR TITLE
Django v1.8 upgrade

### PR DESCRIPTION
**Pull request by symbolist**
https://openedx.atlassian.net/browse/TNL-2818

Originally reviewed in https://github.com/edx/edx-ora2/pull/721.

### CODE REVIEW

File: openassessment/fileupload/backends/filesystem.py [hunk 1]
```diff
@@ -93,13 +94,13 @@ def is_upload_url_available(url_key_name):
...
     """
-    return get_cache().get(get_upload_cache_key(url_key_name)) is not None
+    return get_cache().get(smart_text(get_upload_cache_key(url_key_name))) is not None
 
 def is_download_url_available(url_key_name):
     """
     Return True if the corresponding download URL is available.
     """
-    return get_cache().get(get_download_cache_key(url_key_name)) is not None
+    return get_cache().get(smart_text(get_download_cache_key(url_key_name))) is not None
```
Comment (symbolist) [inline openassessment/fileupload/backends/filesystem.py]: @wedaly I have made some changes after your feedback. A few notes:
1. These changes do not the effect the s3 backend at all. 
2. The issue that was causing failures was that the Django v1.8 cache methods try to decode byte-strings with ascii encoding. The traceback I was seeing for `test_upload_download_with_accented_key()` is copied below. So one solution I can think of is to convert the bytestring keys to unicode explicitly before passing them to the cache methods.

```
======================================================================
ERROR: test_upload_download_with_accented_key (test_api.TestFileUploadServiceWithFilesystemBackend)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/edx/app/edxapp/lib/ora2/openassessment/fileupload/tests/test_api.py", line 275, in test_upload_download_with_accented_key
    upload_response = self.client.put(upload_url, data=self.content.read(), content_type=self.content_type)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/test/client.py", line 546, in put
    secure=secure, **extra)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/test/client.py", line 339, in put
    secure=secure, **extra)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/test/client.py", line 379, in generic
    return self.request(**r)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/test/client.py", line 466, in request
    six.reraise(*exc_info)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/http.py", line 45, in inner
    return func(request, *args, **kwargs)
  File "/edx/app/edxapp/lib/ora2/openassessment/fileupload/views_filesystem.py", line 23, in filesystem_storage
    if not is_upload_url_available(key):
  File "/edx/app/edxapp/lib/ora2/openassessment/fileupload/backends/filesystem.py", line 96, in is_upload_url_available
    return get_cache().get(get_upload_cache_key(url_key_name)) is not None
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/cache/backends/locmem.py", line 46, in get
    key = self.make_key(key, version=version)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/cache/backends/base.py", line 102, in make_key
    new_key = self.key_func(key, self.key_prefix, version)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/cache/backends/base.py", line 35, in default_key_func
    return '%s:%s:%s' % (key_prefix, version, key)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 33: ordinal not in range(128)
```

**Comment by wedaly** [comment]
Okay, that makes sense to me.  Decoding the bytestring back to unicode before passing it to the cache seems reasonable.

**Comment by symbolist** [comment]
@wedaly Do you need to review this further or are you done?

**Comment by wedaly** [comment]
:+1:

**Comment by nedbat** [comment]
This needs to bump the version number in setup.py

**Pull request closed without merging**